components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -302,7 +302,9 @@ func (wsc *WorkspaceController) handleWorkspaceStop(ctx context.Context, ws *wor
 		wsc.metrics.recordFinalizeTime(time.Since(disposeStart).Seconds(), ws)
 	}
 
-	wsc.emitEvent(ws, "Backup", fmt.Errorf("failed to backup workspace: %w", disposeErr))
+	if disposeErr != nil {
+		wsc.emitEvent(ws, "Backup", fmt.Errorf("failed to backup workspace: %w", disposeErr))
+	}
```

Comment:
fixes a Backup failure event being generated even on successful backups, as the error is wrapped and therefore always non-nil, so a failure event gets generated